erpnext/accounts/doctype/sales_invoice/sales_invoice.js

```diff
@@ -820,6 +820,12 @@ frappe.ui.form.on('Sales Invoice', {
 			method: "erpnext.accounts.doctype.sales_invoice.sales_invoice.create_invoice_discounting",
 			frm: frm
 		});
+	},
+	order_type: function(frm) {
+		let discount_percentage = 100;
+		if(frm.doc.order_type === "Marketing") {
+			frm.set_value("additional_discount_percentage", discount_percentage);
+		}
 	}
```

Comment:
If we want to apply this for all doctypes, put this in the `sales_common.js` file instead

Also, handle the case where the percentage goes back to 0 if something else is selected (check the Sample order logic)